### ISSUE TITLE
BUG add blogpost button fails when site in subfolder

### ISSUE THREAD

**Issue by mikeyc7m** [silverstripe/silverstripe-blog#276]
I've installed silverstripe in a subfolder but serving the site from a top-level URL. Everything works except the "add new blog post" button, it throws a 500 error. I can get around the issue by going to the site tree, right-click on the blog and add a new page. Not sure why the button fails to redirect to a new post (could be a lumberjack issue?) but here is where it dies:
http://www...com.au/admin/pages/edit/EditForm/field/ChildPages?ChildPages%5BGridState%5D=%7B%22GridFieldSortableHeader%22%3A%7B%22SortColumn%22%3A%5B%5D%7D%2C%22GridFieldFilterHeader%22%3A%7B%22Columns%22%3A%5B%5D%7D%2C%22GridFieldPaginator%22%3A%7B%22currentPage%22%3A1%7D%2C%22GridFieldSiteTreeAddNewButton%22%3A%7B%22currentPageID%22%3A28%2C%22pageType%22%3A%22BlogPost%22%7D%7D&Categories%5BGridState%5D=%7B%22GridFieldSortableHeader%22%3A%7B%22SortColumn%22%3A%5B%5D%7D%2C%22GridFieldFilterHeader%22%3A%7B%22Columns%22%3A%5B%5D%7D%2C%22GridFieldPaginator%22%3A%7B%22currentPage%22%3A1%7D%7D&Tags%5BGridState%5D=%7B%22GridFieldSortableHeader%22%3A%7B%22SortColumn%22%3A%5B%5D%7D%2C%22GridFieldFilterHeader%22%3A%7B%22Columns%22%3A%5B%5D%7D%2C%22GridFieldPaginator%22%3A%7B%22currentPage%22%3A1%7D%7D&PageBanners%5BGridState%5D=%7B%22GridFieldOrderableRows%22%3A%7B%22enabled%22%3Atrue%7D%2C%22GridFieldRelationHandler%22%3A%7B%22RelationVal%22%3A%5B%5D%2C%22FirstTime%22%3A1%2C%22ShowingRelation%22%3A0%7D%2C%22GridFieldSortableHeader%22%3A%7B%22SortColumn%22%3A%5B%5D%7D%2C%22GridFieldFilterHeader%22%3A%7B%22Columns%22%3A%5B%5D%7D%2C%22GridFieldPaginator%22%3A%7B%22currentPage%22%3A1%7D%7D&action_gridFieldAlterAction%3FStateID%3D3ac21a873ce600b8b6fe34da2fa5176d=Add%20new%20Blog%20Post

**Comment by dhensby** [comment]
Please can you give us a bit more info about your environment?

 How are you serving from a sub folder to a root url? 

What error do you get back?

**Comment by mikeyc7m** [comment]
ok so my ./htaccess is like:
````
RewriteBase '/'
RewriteCond %{REQUEST_URI} !^/silverstripe/
RewriteRule ^(.*)$ /silverstripe/$1 [L]
````
and my /silverstripe/.htaccess is all like:
````
RewriteBase '/silverstripe/'
RewriteCond %{REQUEST_URI} ^(.*)$
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_URI} !\.php$
RewriteRule .* framework/main.php?url=%1&%{QUERY_STRING} [L]
````
and my /silverstripe/mysite/_config/config.yml is like yeah man:
````
Director:
  alternate_base_url: '/'
````
and my error page is all like "500 server error" and stuff. Doesn't hit the target code to create the post, just dies on the redirect.

**Comment by dhensby** [comment]
Great, thanks. Lastly, can you confirm the versions of SS and blog module in use?

**Comment by mikeyc7m** [comment]
SS 3.1.9
blog & lumberjack is dev-master, maybe a week old.

**Comment by micmania1** [comment]
Its probably related to the link being generated here: https://github.com/silverstripe/silverstripe-lumberjack/blob/1.1/code/forms/gridfield/GridFieldSiteTreeAddNewButton.php#L133

Can you try debugging this? What does Director::absoluteBaseURL() return?

**Comment by mikeyc7m** [comment]
yeah that was my original guess too. i can't debug, it's a live site, but no new record is generated so i'm thinking it occurs sooner. ok well maybe i might try sneaking in a debug...

**Comment by mikeyc7m** [comment]
yeah no good, the handleAction method isn't reached - the button's url is a dead link.

**Comment by mikeyc7m** [comment]
Great Success! The error was occurring before handleAction and throwing a 500 error, so I checked the php config. Turns out the php suhosin extension chokes on URLs longer than 512 chars. Removed it and the "add new blog post" button is working fine.

**Comment by dhensby** [comment]
Ok. Related: https://github.com/silverstripe/silverstripe-framework/issues/3855

I believe this is fixed in a new version of ss

These overly long URLs are something we should look into, I'm not sure this data should be sent as GET cars...

**Comment by micmania1** [comment]
Yeah, I think this was fixed in 3.1.10. @mikeyc7m can you try upgrading?

**Comment by micmania1** [comment]
This looks to have been fixed in core and there hasn't been a response in almost a year so i'm closing. Feel free to re-open if this issue isn't solved.
